### PR TITLE
Ability to select media type

### PR DESCRIPTION
**Pull request by prin53**
In this PR you can specify media type you need when creating `ChafuViewController`:

```
new ChafuViewController
{
    LazyDataSource = (albumView, size, mediaTypes) => new PhotoGalleryDataSource(albumView, size, MediaType.Photo | MediaType.Video)
};
```

The default data source will have the same behavior: to show both photo and video.

### CODE REVIEW

File: Chafu/PhotoGalleryDataSource.cs [hunk 1]
```diff
@@ -46,11 +47,12 @@ public class PhotoGalleryDataSource : BaseAlbumDataSource, IPHPhotoLibraryChange
         /// </summary>
         /// <param name="albumView"><see cref="AlbumView"/> attached to</param>
         /// <param name="cellSize"><see cref="CGSize"/> with cell size. If <see cref="CGSize.Empty"/> it will default to 100x100</param>
-        public PhotoGalleryDataSource(AlbumView albumView, CGSize cellSize)
+		public PhotoGalleryDataSource(AlbumView albumView, CGSize cellSize, MediaType mediaTypes)
```
Comment (Cheesebaron) [inline Chafu/PhotoGalleryDataSource.cs]: Please add default value to the ctor and add to the comment above.

Convert tab to 4 spaces

File: Chafu/PhotoGalleryDataSource.cs [hunk 2]
```diff
@@ -46,11 +47,12 @@ public class PhotoGalleryDataSource : BaseAlbumDataSource, IPHPhotoLibraryChange
...
         {
             _albumView = albumView;
             _cellSize = cellSize != CGSize.Empty ? cellSize : new CGSize(100, 100);
             _scale = UIScreen.MainScreen.Scale;
+			_mediaTypes = mediaTypes;
```
Comment (Cheesebaron) [inline Chafu/PhotoGalleryDataSource.cs]: Convert tab to 4 spaces

File: Chafu/PhotoGalleryDataSource.cs [hunk 3]
```diff
@@ -66,12 +68,20 @@ private void OnAuthorized()
                     SortDescriptors = new[] {new NSSortDescriptor("creationDate", false)}
                 };
 
-                _images = PHAsset.FetchAssets(PHAssetMediaType.Image, options);
-                _videos = PHAsset.FetchAssets(PHAssetMediaType.Video, options);
+				var assets = new List<PHAsset>();
```
Comment (Cheesebaron) [inline Chafu/PhotoGalleryDataSource.cs]: Convert tab to 4 spaces

File: Chafu/PhotoGalleryDataSource.cs [hunk 4]
```diff
@@ -125,21 +135,37 @@ public void PhotoLibraryDidChange(PHChange changeInstance)
...
+						_images = AdjustAssets(_videos, videoCollectionChanges);
+					}
+				}
             });
         }
 
         private PHFetchResult AdjustAssets(PHFetchResult assets, PHFetchResultChangeDetails changes)
         {
-            var before = assets;
+			if (assets == null)
```
Comment (Cheesebaron) [inline Chafu/PhotoGalleryDataSource.cs]: Convert tab to 4 spaces

File: Chafu/PhotoGalleryDataSource.cs [hunk 5]
```diff
@@ -125,21 +135,37 @@ public void PhotoLibraryDidChange(PHChange changeInstance)
...
+			{
```
Comment (Cheesebaron) [inline Chafu/PhotoGalleryDataSource.cs]: Remove { } around statement

File: Chafu/PhotoGalleryDataSource.cs [hunk 6]
```diff
@@ -125,21 +135,37 @@ public void PhotoLibraryDidChange(PHChange changeInstance)
...
-                var imageCollectionChanges = changeInstance.GetFetchResultChangeDetails(_images);
-                var videoCollectionChanges = changeInstance.GetFetchResultChangeDetails(_videos);
-
-                if (imageCollectionChanges != null)
-                    _images = AdjustAssets(_images, imageCollectionChanges);
-
-                if (videoCollectionChanges != null)
-                    _videos = AdjustAssets(_videos, videoCollectionChanges);
+				//var collectionView = _albumView.CollectionView;
+				if (_images != null)
```
Comment (Cheesebaron) [inline Chafu/PhotoGalleryDataSource.cs]: Split out in own method.

Convert tab to 4 spaces.

File: Chafu/MediaType.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Chafu
+{
+	[Flags]
+	public enum MediaType
```
Comment (Cheesebaron) [inline Chafu/MediaType.cs]: Convert tab to 4 spaces

Comment (Cheesebaron) [reply Chafu/MediaType.cs]: There is already a MediaType class here: https://github.com/BruelAndKjaer/Chafu/blob/a51cbbcf0a5c70c0e2d08f7e020129e15fdb3aed/Chafu/BaseAlbumDataSource.cs#L9 use that

File: Chafu/AlbumViewController.cs [hunk 8]
```diff
@@ -110,7 +110,7 @@ private void CreateViews()
                 BackgroundColor = Configuration.BackgroundColor
             };
 
-            AlbumDataSource = LazyDataSource(_album, CellSize);
+			AlbumDataSource = LazyDataSource(_album, CellSize, MediaType.Photo | MediaType.Video);
```
Comment (Cheesebaron) [inline Chafu/AlbumViewController.cs]: Convert tabs to 4 spaces each

**Review by Cheesebaron — CHANGES_REQUESTED** [review]
Thank you for this pull request. Please fix the requested changes commented per file.

In general, convert tabs to 4 spaces.

**Comment by prin53** [comment]
@Cheesebaron thank you for your comments. I pushed related changes.

**Review by Cheesebaron — APPROVED** [review]
Looks good 👍

**Comment by Cheesebaron** [comment]
Will give it a spin when I have access to an iPhone again.

**Comment by Cheesebaron** [comment]
This is out in the 1.5.0 NuGet. Thanks 👍

**Pull request merged**